src/configurator.cpp

```diff
@@ -78,6 +89,27 @@ void Configurator::initCcnet()
 
 void Configurator::initSeafile()
 {
+#if defined(Q_OS_WIN32)
+    QString preconfigure_dir = getPreconfigureDirectory();
+    if (!preconfigure_dir.isEmpty()) {
+        QDir dir(preconfigure_dir);
+        if (!dir.mkpath(".") ||
+            !dir.mkpath("Seafile/.seafile-data")) {
+            QMessageBox::warning(NULL, getBrand(),
+                                 tr("Unable to create preconfigure directory"),
+                                 QMessageBox::Ok);
+            exit(1);
```

Comment:
When we failed to create the foder whose path is pre-configured in the registry, we have two options:

- show an error box and exit (as this PR does)
- ask the user to choose a directory (as we used to).

I'm not sure which one is better.